cms/djangoapps/contentstore/management/commands/swap_from_auto_track_cohort_pilot.py

```diff
@@ -0,0 +1,168 @@
+                        set_audit_enrollment_track = True
+
+                # Check the partition and group IDs for the verified course group, if it exists in the xblock's access
+                # settings then set the verified track flag to true
+                verified_partition_group_access = item.group_access.get(
+                    verified_course_user_group_partition_group.partition_id,
+                    None
+                )
+                if (verified_partition_group_access
+                        and verified_course_user_group_partition_group.group_id in verified_partition_group_access):
+                    set_verified_enrollment_track = True
+
+                # Add the enrollment track ids to a group access array
+                enrollment_track_group_access = []
+                if set_audit_enrollment_track:
+                    enrollment_track_group_access.append(audit_course_mode.id)
+                if set_verified_enrollment_track:
+                    enrollment_track_group_access.append(verified_course_mode.id)
+
+                # If either the audit track, or verified track needed to be updated set the access, update and publish
+                if set_verified_enrollment_track or set_audit_enrollment_track:
+                    # Check that the xblock is published if it needs changes, otherwise raise an error
+                    if not published:
+                        raise CommandError('XBlock for course, %s needs access changes, but is a draft' % course_id)
+                    item.group_access = {ENROLLMENT_TRACK_PARTITION_ID: enrollment_track_group_access}
+                    module_store.update_item(item, ModuleStoreEnum.UserID.mgmt_command)
+                    module_store.publish(item.location, ModuleStoreEnum.UserID.mgmt_command)
+
+        partitions_to_delete = random_audit_course_user_group_partition_groups
+        partitions_to_delete.append(verified_course_user_group_partition_group)
+
+        # Check if we should delete any partition groups,
+        # Taken from contentstore/views/course.py.remove_content_or_experiment_group
+        if partitions_to_delete:
+            partition_service = PartitionService(course_key)
+            course = partition_service.get_course()
+            for partition_to_delete in partitions_to_delete:
+                # Get the user partition, and the index of that partition in the course
+                partition = partition_service.get_user_partition(partition_to_delete.partition_id)
+                partition_index = course.user_partitions.index(partition)
+                group_id = int(partition_to_delete.group_id)
+
+                # Sanity check to verify that all of the groups being deleted are empty,
+                # since they should have been converted to use enrollment tracks instead
+                usages = GroupConfiguration.get_partitions_usage_info(module_store, course)
```

Comment:
Running for the demo course it returns ```
{1: [{'url': u'/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@5f514c9693764f65848f6b634dd15f67', 'label': u'Demo Course Overview / Unit'}, {'url': u'/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@77e5455f4d664c2e91af710a08538bd0', 'label': u'Demo Course Overview / Unit'}], 2078571546: [{'url': u'/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@vertical_0270f6de40fc', 'label': u'Demo Course Overview / Introduction: Video and Sequences'}], 2: [{'url': u'/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@5f514c9693764f65848f6b634dd15f67', 'label': u'Demo Course Overview / Unit'}, {'url': u'/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@8f18f19cc65843dd947ac54bdb319e0e', 'label': u'Demo Course Overview / Unit'}, {'url': u'/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@43c00efa607b42239114a34f9887e630', 'label': u'Demo Course Overview / Visible to Group A'}]}
```

So it doesn't really make sense to be checking the group ID against the partition IDs here. I looked at this code in https://github.com/edx/edx-platform/blob/master/cms/djangoapps/contentstore/views/course.py#L1493 that seemed to be doing a similar check that was needed on whether or not a group was in use.